### PR TITLE
Feat: New rule: samtools_stats for general qc using samtools

### PR DESCRIPTION
**Pull request by jonca79**

### CODE REVIEW

File: .tests/integration/config.yaml [hunk 1]
```diff
@@ -2,11 +2,15 @@ resources: "resources.yaml"
...
 reference:
   fasta: "reference/HES45.fna"
   design_intervals: "reference/twist_DNA_solid.HES45.annotated.interval_list"
+  design_bed: "reference/twist_DNA_solid.HES45.annotated.bed"
 
 fastqc:
   container: "docker://hydragenetics/fastqc:0.11.9"
+
+samtools_stats:
+  extra: "-t reference/twist_DNA_solid.HES45.annotated.bed"
```
Comment (jonca79) [inline .tests/integration/config.yaml:16]: Hur gör man här? Skulle vilja ha: 
extra: "-t %s" % config[reference][design_bed]
men det fungerar väl inte?

Comment (Smeds) [reply .tests/integration/config.yaml:16]: det borde fungera, eller?

File: .tests/integration/config.yaml [hunk 2]
```diff
@@ -2,11 +2,15 @@ resources: "resources.yaml"
 samples: "samples.tsv"
 units: "units.tsv"
 
-default_container: "docker://hydragenetics/common:0.0.1"
+default_container: "docker://gmsuppsala/somatic:develop"
```
Comment (Smeds) [inline .tests/integration/config.yaml:5]: you should be able to use our common, it now contains python

File: config/config.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@ resources: "resources.yaml"
 samples: "samples.tsv"
 units: "units.tsv"
 
-default_container: "docker://hydragenetics/common:0.0.1"
+default_container: "docker://gmsuppsala/somatic:develop"
```
Comment (Smeds) [inline config/config.yaml:5]: you should be able to use our common, it now contains python

File: workflow/envs/samtools_stats.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+channels:
+  - conda-forge
+  - bioconda
+  - defaults
+dependencies:
+  - samtools==1.10
```
Comment (Smeds) [inline workflow/envs/samtools_stats.yaml:6]: our container has version 1.14, should we change to same here?
```suggestion
  - samtools==1.14
```

**Review by Smeds — CHANGES_REQUESTED** [review]

File: workflow/envs/samtools_stats.yaml (hunk 4 above)
Comment (jonca79) [reply workflow/envs/samtools_stats.yaml:6]: It is from the wrapper but I guess it should be ok to change

File: .tests/integration/config.yaml [hunk 5]
```diff
@@ -7,18 +7,23 @@ default_container: "docker://hydragenetics/common:0.0.1"
...
   design_intervals: "reference/twist_DNA_solid.HES45.annotated.interval_list"
+  design_bed: "reference/twist_DNA_solid.HES45.annotated.bed"
 
 fastqc:
   container: "docker://hydragenetics/fastqc:0.11.9"
-  
+
+samtools_stats:
+  extra: "-t reference/twist_DNA_solid.HES45.annotated.bed"
+  container: "docker://gmsuppsala/somatic:develop"
```
Comment (Smeds) [inline .tests/integration/config.yaml:17]:
```suggestion
  container: "docker://hydragenetics/common:0.0.1"
```

**Review by Smeds — APPROVED** [review]

**Pull request merged**